src/events/cd-event-list.vue

```diff
@@ -54,12 +66,16 @@
         this.events = res.body;
       },
       async joinTheDojo() {
-        const userType = UsersUtil.isYouthOverThirteen(new Date(this.usersProfile.dob)) ? 'attendee-o13' : 'parent-guardian';
-        await DojosService.joinDojo(this.currentUser.id, this.dojo.id, [userType]);
-        /* eslint-disable no-alert */
-        alert('Congratulations, you have now joined the Dojo.');
-        /* eslint-enable no-alert */
-        this.loadUserDojoRole();
+        if (this.currentUser) {
+          const userType = UsersUtil.isYouthOverThirteen(new Date(this.usersProfile.dob)) ? 'attendee-o13' : 'parent-guardian';
+          await DojosService.joinDojo(this.currentUser.id, this.dojo.id, [userType]);
+          /* eslint-disable no-alert */
+          alert('Congratulations, you have now joined the Dojo.');
+          /* eslint-enable no-alert */
+          this.loadUserDojoRole();
+        } else {
+          location.href = `/register/account?referer=${this.$route.path}`;
```

Comment:
why register over login?

Reply:
Because this page is the one displaying both register and login